pkg/controller/controller.go

```diff
@@ -221,39 +222,43 @@ func (c *Controller) FindResourceVersion(tprClient *rest.RESTClient) (string, er
 }
 
 // monitor continuously watches for changes to custom function objects
-func (c *Controller) monitor(httpClient *http.Client, watchVersion string) (<-chan *Event, <-chan error) {
+func (c *Controller) monitor(httpClient *http.Client, watchVersion string) <-chan *Event {
 	eventCh := make(chan *Event)
-	// keep the latest unexpected error only
-	var err error
 
 	go func() {
 		defer close(eventCh)
+
+		// per-watch loop: start watching resources and collecting functions (custom objects)
 		for {
 			resp, err := utils.WatchResources(httpClient, watchVersion)
 			if err != nil {
+				resp.Body.Close()
```

Comment:
`resp` is probably undefined here.  We don't need to close the Body in the `err != nil` case (it's up to the internals of `utils.WatchResources` to not leak in the error case)